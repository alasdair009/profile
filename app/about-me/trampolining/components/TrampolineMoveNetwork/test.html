<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
