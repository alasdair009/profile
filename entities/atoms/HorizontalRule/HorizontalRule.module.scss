@use "../../../app/styles/animation";
@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";

.root {
  border: none;
  margin: rem.toRem(map-get(dimensions.$sizes, s32)) auto;
  max-width: rem.toRem(dimensions.$globalDecorationMaxWidth);
  padding: rem.toRem(1);
  position: relative;
  width: 100%;
  z-index: 1;

  &::before {
    background: colors.$colorGreenGrass;
    content: "";
    height: rem.toRem(2);
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 100%;
  }

  &::after {
    aspect-ratio: 1;
    background: colors.$colorblackEvil url(../../assets/am.svg) no-repeat center;
    border: rem.toRem(2) solid colors.$colorGreenGrass;
    border-radius: 50%;
    content: "";
    display: none;
    height: rem.toRem(map-get(dimensions.$sizes, s48));
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    transition: transform animation.animationDuration(1);

    &:hover {
      transform: translate(-50%, -50%) rotateX(360deg);
    }
  }

  &.Top {
    margin: rem.toRem(map-get(dimensions.$sizes, s32)) auto 0;
  }
  &.Bottom {
    margin: 0 auto rem.toRem(map-get(dimensions.$sizes, s32));
  }
  &.None {
    margin: 0 auto;
  }

  &.Decoration {
    padding: calc(rem.toRem(map-get(dimensions.$sizes, s24)) + rem.toRem(2));

    &::after {
      display: block;
    }

    &.Top {
      margin-bottom: calc(
        rem.toRem(map-get(dimensions.$sizes, s24)) - rem.toRem(2) * -1
      );
    }

    &.Bottom {
      margin-top: calc(
        rem.toRem(map-get(dimensions.$sizes, s24)) - rem.toRem(2) * -1
      );
    }

    &.None {
      margin: calc(
          rem.toRem(map-get(dimensions.$sizes, s24)) - rem.toRem(2) * -1
        )
        auto;
    }
  }
}
