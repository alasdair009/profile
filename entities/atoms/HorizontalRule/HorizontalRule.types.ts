export type HorizontalRuleMarginOptions = "top" | "bottom" | "both" | "none";