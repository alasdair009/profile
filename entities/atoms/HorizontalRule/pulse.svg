<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 600 30">
    <defs>
        <style>
            .path {
                fill: transparent;
                stroke: white;
                stroke-width: 1px;
            }
        </style>
    </defs>
    <path class="path" d="M 0,20 l 270,0 c 20,-20 40,-20 60,0 l 270,0" />
</svg>