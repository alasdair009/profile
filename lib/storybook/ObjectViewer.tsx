import { Source } from "@storybook/blocks";

import type { JSX } from "react";

export type ObjectViewerProps = {
  /**
   * The object to display
   */
  object: object;
  /**
   * The title of the code segment
   */
  title: string;
};

export function ObjectViewer({
  object,
  title,
}: ObjectViewerProps): JSX.Element {
  // Format the object as JSON with indentation
  const objectAsString = JSON.stringify(object, null, 4);

  return (
    <div title={title} data-testid={ObjectViewer.name}>
      <Source code={objectAsString} dark={true} />
    </div>
  );
}
