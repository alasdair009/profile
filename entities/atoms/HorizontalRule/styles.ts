"use client";
import { rem } from "polished";

import { HorizontalRuleMarginOptions } from "./HorizontalRule.types";
import { colors, globalDecorationMaxWidth, sizes } from "@/entities";
import styled from "styled-components";
import amLogo from "../../assets/am.svg";

const getMarginStyle = (
  margin: HorizontalRuleMarginOptions,
  decoration: boolean
): string => {
  switch (margin) {
    case "top":
      return `${sizes.s32.rem} auto ${decoration ? `calc(-${sizes.s24.rem} - ${rem(2)})` : 0}`;
    case "bottom":
      return `${decoration ? `calc(-${sizes.s24.rem} - ${rem(2)})` : 0} auto ${sizes.s32.rem}`;
    case "both":
      return `${sizes.s32.rem} auto`;
    case "none":
      return decoration ? `calc(-${sizes.s24.rem} - ${rem(2)}) auto` : `0 auto`;
  }
};
export const Root = styled.hr<{
  $decoration: boolean;
  $margin: HorizontalRuleMarginOptions;
}>`
  border: none;
  margin: ${({ $margin, $decoration }) => getMarginStyle($margin, $decoration)};
  max-width: ${rem(globalDecorationMaxWidth)};
  padding: ${({ $decoration }) =>
      $decoration ? `calc(${sizes.s24.rem} + ${rem(2)})` : rem(1)}
    0;
  position: relative;
  width: 100%;
  z-index: 1;

  &::before {
    background: ${colors.greenGrass};
    content: "";
    height: ${rem(2)};
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 100%;
  }

  &::after {
    aspect-ratio: 1;
    background: ${colors.blackEvil} url(${amLogo.src}) no-repeat center;
    border: ${rem(2)} solid ${colors.greenGrass};
    border-radius: 50%;
    content: "";
    display: ${({ $decoration }) => ($decoration ? "block" : "none")};
    height: ${sizes.s48.rem};
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    transition: transform 1s;

    &:hover {
      transform: translate(-50%, -50%) rotateX(360deg);
    }
  }
`;
