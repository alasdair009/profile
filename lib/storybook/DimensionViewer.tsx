import { Source } from "@storybook/blocks";
import type { HTMLAttributes } from "react";

import { ObjectViewer } from "./ObjectViewer";
import { colors } from "@/entities";

export type DimensionViewerProps = {
  /**
   * The units to display
   */
  units: Record<string, unknown>;
  /**
   * A simplified object that can be used for the DimensionBar interface
   */
  barUnits: Record<string, string | number>;
  /**
   * The title of the code segment
   */
  title: string;
  /**
   * The suffix to display after the unit
   */
  suffix?: string;
  /**
   * Source example
   */
  sourceExample: string;
  /**
   * The name of the object
   */
  objectName: string;
};

export function DimensionViewer({
  units,
  title,
  sourceExample,
  objectName,
  barUnits,
  suffix = "",
}: DimensionViewerProps): JSX.Element {
  return (
    <div data-testid={DimensionViewer.name}>
      <ObjectViewer object={units} title={objectName} />
      <p>
        To use these values import the <strong>{objectName}</strong> object and
        reference the required key.
      </p>
      <Source code={sourceExample} dark={true} language="css" />
      <DimensionBarWrapper title={title}>
        {Object.keys(barUnits).map((key, index) => {
          return (
            <DimensionBar
              key={`${title}${index}`}
              data-testid={DimensionViewer.name}
              width={`${barUnits[key]}${suffix}`}
            >
              <DimensionBarLabel>
                {barUnits[key]}
                {suffix}
                <br />({key})
              </DimensionBarLabel>
            </DimensionBar>
          );
        })}
      </DimensionBarWrapper>
    </div>
  );
}

const DimensionBarWrapper = ({ children }: HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      overflow: "scroll",
      width: "100%",
    }}
  >
    {children}
  </div>
);

const DimensionBar = ({
  children,
  width,
}: { width: string } & HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      background: colors.greenGrass,
      color: "white",
      flex: 1,
      marginBottom: "8px",
      width,
    }}
  >
    {children}
  </div>
);

const DimensionBarLabel = ({ children }: HTMLAttributes<HTMLSpanElement>) => (
  <span
    style={{
      display: "block",
      marginLeft: "8px",
    }}
  >
    {children}
  </span>
);
