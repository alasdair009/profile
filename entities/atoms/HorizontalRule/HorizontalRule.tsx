import { Root } from "./styles";
import { HTMLAttributes } from "react";
import {HorizontalRuleMarginOptions} from "@/entities/atoms/HorizontalRule/HorizontalRule.types";

type HorizontalRuleProps = {
    decoration?: boolean;
    margin?: HorizontalRuleMarginOptions;
} & HTMLAttributes<HTMLHRElement>;

export function HorizontalRule({ decoration = false, margin = "both", ...rest }: HorizontalRuleProps) {
    return <Root $decoration={decoration} $margin={margin} {...rest} />;
}
