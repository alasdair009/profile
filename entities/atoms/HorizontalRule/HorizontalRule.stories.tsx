import { HorizontalRule } from "./HorizontalRule";
import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, within } from "storybook/test";

const meta: Meta<typeof HorizontalRule> = {
  component: HorizontalRule,
  args: {
    curve: false,
    decoration: false,
    margin: "both",
  },
};
export default meta;

export const Default: StoryObj<typeof HorizontalRule> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const ruleElement = canvas.getByTestId(HorizontalRule.name);

    await expect(ruleElement).toBeInTheDocument();
  },
};

export const Decoration: StoryObj<typeof HorizontalRule> = {
  args: {
    decoration: true,
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const ruleElement = canvas.getByTestId(HorizontalRule.name);

    await expect(ruleElement).toBeInTheDocument();
  },
};

export const Curve: StoryObj<typeof HorizontalRule> = {
  args: {
    curve: true,
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const ruleElement = canvas.getByTestId(HorizontalRule.name);

    await expect(ruleElement).toBeInTheDocument();
  },
};
