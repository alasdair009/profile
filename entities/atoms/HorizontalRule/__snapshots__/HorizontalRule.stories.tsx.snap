// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/HorizontalRule Decoration play-test 1`] = `
<hr data-testid="HorizontalRule"
    class="sc-hDcvty kbmuZx"
>
`;

exports[`atoms/HorizontalRule Default play-test 1`] = `
<hr data-testid="HorizontalRule"
    class="sc-hDcvty cbbhpu"
>
`;
